File: task1.java
import java.util.Scanner;

public class ReverseSubstring {
    public static void main(String[] args) {
      
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a paragraph: ");
        String paragraph = scanner.nextLine();

 
        String reversedParagraph = reverseSubstring(paragraph);

        System.out.println(reversedParagraph);
    }

    public static String reverseSubstring(String s) {
        if (s.length() <= 1) {
            return s;
        } else {
            int mid = s.length() / 2;
            String left = s.substring(0, mid);
            String right = s.substring(mid);
           
            Thread leftThread = new Thread(() -> {
                reverseSubstring(left);
            });
            Thread rightThread = new Thread(() -> {
                reverseSubstring(right);
            });
            leftThread.start();
            rightThread.start();
            try {
                
                leftThread.join();
                rightThread.join();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
           
            return new StringBuilder(right).reverse().toString() +
                   new

File: task1.py
import threading

def reverse_substring(s):
    if len(s) <= 1:
        return s
    else:
        mid = len(s) // 2
        left = s[:mid]
        right = s[mid:]
      
        left_thread = threading.Thread(target=reverse_substring, args=(left,))
        right_thread = threading.Thread(target=reverse_substring, args=(right,))
        left_thread.start()
        right_thread.start()
      
        left_thread.join()
        right_thread.join()
        
        return right[::-1] + left[::-1]


paragraph = input("Enter a paragraph: ")


reversed_paragraph = reverse_substring(paragraph)

print(reversed_paragraph)
